### PR TITLE
#153: Added current location to search results list

### PR DESCRIPTION
**Pull request by narrapangan**
Current location will be added at the top of the list of search results. I found this solution to be better than having a designated button on the search activity page because it will only be displayed if the current location is available. 

Search Activity Page
**Note: Will Show on Map be implemented?**
Screen Shot 2020-04-04 at 8 01 51 AM

Once user taps on "Search Here", the search results (and current location) will be displayed
Screen Shot 2020-04-04 at 8 02 03 AM

Once user taps on "My Location: xxxx, xxxx", the current location is populated 
Screen Shot 2020-04-04 at 8 04 54 AM

### CODE REVIEW

**Comment by ferdousara-parvin** [comment]
Can we not show the lat and long tho?

**Comment by ferdousara-parvin** [comment]
Also to answer your question "Will Show on Map be implemented?" --> Answer is no so can you remove it please?

**Comment by narrapangan** [comment]
@ferdousara-parvin It was what was set as default in the my current place. I assumed we wanted it that way, I will change it. And I will remove the Show on Map. 

Thanks!

**Comment by narrapangan** [comment]
Show on Map is removed. (However seems quite plain, any ideas on what we could add?)
image

Coordinates were removed! 
image

File: app/src/main/java/com/example/concordia_campus_guide/Models/MyCurrentPlace.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
     public MyCurrentPlace(Double latitude, Double longitude){
         this.setCenterCoordinates(new Coordinates(latitude, longitude));
-        this.displayName = "My location: " + latitude.toString() + ", " + longitude.toString();
+        this.displayName = "My Current Location";
```
Comment (tarekait1996) [inline app/src/main/java/com/example/concordia_campus_guide/Models/MyCurrentPlace.java:9]: use string resource, please do not hard code the string anymore

**Comment by thomasgauvin** [comment]
I think it was ok to show the lat lng though

**Comment by thomasgauvin** [comment]
https://github.com/Concordia-Campus-Guide/Concordia-Campus-Guide/pull/242

check this out Narra, this is what I was talking about

**Comment by narrapangan** [comment]
@thomasgauvin I was thinking of doing it like that, however once the user taps on the search bar, there is no way of going back to that initial view.

**Comment by ferdousara-parvin** [comment]
> 
> 
> I think it was better to show the lat lng though

But it's ugly no? Like a real user doesn't really care.

File: app/src/main/java/com/example/concordia_campus_guide/Models/MyCurrentPlace.java (hunk 1 above)
Comment (narrapangan) [reply app/src/main/java/com/example/concordia_campus_guide/Models/MyCurrentPlace.java]: I updated that (look below)

**Comment by narrapangan** [comment]
image

File: app/src/main/java/com/example/concordia_campus_guide/Models/MyCurrentPlace.java [hunk 2]
```diff
@@ -1,14 +1,20 @@
...
+import android.content.Context;
+
+import com.example.concordia_campus_guide.R;
+
 public class MyCurrentPlace extends Place {
-    String displayName = "Select location";
+    String displayName = "";
 
-    public MyCurrentPlace(Double latitude, Double longitude){
+    public MyCurrentPlace(Context context, Double longitude, Double latitude){
```
Comment (narrapangan) [inline app/src/main/java/com/example/concordia_campus_guide/Models/MyCurrentPlace.java:10]: Context had to be added to access string resource

File: app/src/test/java/com/example/concordia_campus_guide/ModelTests/MyCurrentPlaceTest.java [hunk 3]
```diff
@@ -12,13 +12,14 @@
 
     @Before
     public void init() {
-        currentPlace = new MyCurrentPlace(-73.57907921075821, 45.49702057370776);
+        //TODO: add context
```
Comment (narrapangan) [inline app/src/test/java/com/example/concordia_campus_guide/ModelTests/MyCurrentPlaceTest.java:15]: Will most likely end up doing instrumented test for this

**Comment by thomasgauvin** [comment]
🔥

**Pull request merged**